bugbear.py

```diff
@@ -68,24 +68,29 @@ def gen_line_based_checks(self):
 
         The following simple checks are based on the raw lines, not the AST.
         """
+        noqa_type_ignore_regex = re.compile(r"#\s*(noqa|type:\s*ignore)[^#\r\n]*$")
         for lineno, line in enumerate(self.lines, start=1):
             # Special case: ignore long shebang (following pycodestyle).
             if lineno == 1 and line.startswith("#!"):
                 continue
 
-            length = len(line) - 1
-            if length > 1.1 * self.max_line_length and line.strip():
+            # At first, removing noqa and type: ignore trailing comments"
+            line_without_first_comment = noqa_type_ignore_regex.sub("", line)
+            no_comment_line = noqa_type_ignore_regex.sub("", line_without_first_comment)
```

Comment:
```suggestion
            no_comment_line = noqa_type_ignore_regex.sub("", line)
            if no_comment_line != line:
                no_comment_line = noqa_type_ignore_regex.sub("", no_comment_line)
```

We only need to run a second regex pass if the first one actually changes anything.

Reply:
Changed 👍